internal/brokers/broker.go

```diff
@@ -259,33 +259,40 @@ func (b Broker) validateUILayout(sessionID string, layout map[string]string) (r
 	b.layoutValidatorsMu.Lock()
 	defer b.layoutValidatorsMu.Unlock()
 
-	layoutValidator, exists := b.layoutValidators[sessionID]
+	layoutValidators, exists := b.layoutValidators[sessionID]
 	if !exists {
 		return nil, fmt.Errorf("session %q does not have any layout validator", sessionID)
 	}
 
-	typ := layout["type"]
-	layoutTypeValidator, exists := layoutValidator[typ]
+	layoutValidator, exists := layoutValidators[layout["type"]]
 	if !exists {
-		return nil, fmt.Errorf("no validator for UI layout type %q", typ)
+		return nil, fmt.Errorf("no validator for UI layout type %q", layout["type"])
 	}
 
-	r = map[string]string{"type": typ}
-	for key, validator := range layoutTypeValidator {
+	// Ensure that all provided fields can be validated
+	for key := range layout {
+		if key == "type" {
+			continue
+		}
+		if _, exists := layoutValidator[key]; !exists {
```

Comment:
Maybe add a comment something that the layout is what is provided by broker and the layoutTypeValidator is from the UI (to ensure we understand easilty why we have 2 loops)

Reply:
Isn't [this](https://github.com/ubuntu/authd/blob/d42217e68810c9a73070ba648205fbcf0c99f256/internal/brokers/broker.go#L272) and [this](https://github.com/ubuntu/authd/blob/d42217e68810c9a73070ba648205fbcf0c99f256/internal/brokers/broker.go#L282) enough?

Reply:
maybe it’s a question of variables name, but it was not obvious at first sight to me about the double loop at first, and so I think telling layout is what is coming from the broker, and the layoutValidator are basically the UI constraints help a lot.

Reply:
oof... English is hard. Let me know if it got any better.